docs/src/proposals/secp256k1_instruction.md

```diff
@@ -0,0 +1,75 @@
+---
+title: secp256k1 builtin instruction
+---
+
+## Problem
+
+Need to efficiently perform multiple secp256k1 pubkey recovery operations (ecrecover) in a transaction.
+ecrecover is an ethereum instruction which takes a signature and message and recovers a publickey, a comparison
+to that public key can thus verify that the signature is valid.
+
+The input signature and message data can come in the transaction itself, but there is a requirement for the pubkeys
+to come from an existing account data. Since there needs to be 10-20 signatures in the transaction as well as the
+signing data which is on the order of 500 bytes, transaction space is a concern. But also having more concentrated
+similar work should provide for easier optimization.
+
+## Solution
+
+Add a new builtin instruction which takes in as the first byte a count of the following struct serialized in the instruction
+data:
+
+```
+struct SignatureOffsets {
+    signature_offset: u16,        // offset to [signature,recovery_id] of 64+1 bytes
+    signature_account_index: u8,  // account index of signature data
+    pubkey_offset: u16,           // offset to pubkey in ethereum format of keccak256(pubkey[1..])[12..]
+    pubkey_account_index: u8,     // index of account to receive pubkey
+    message_data_offset: u16,     // offset to start of message data
+    message_data_size: u16,       // size of message data
+    message_account_index: u8,    // index of account to receive message data
+}
+```
+
+Pseudo code of the operation:
+```
+for i in 0..count {
+    // i'th index values referenced:
+    signature = accounts[signature_account_index].data[signature_offset..signature_offset + 64]
```

Comment:
accounts?

Reply:
Yes, the instruction's referenced accounts:
```
pub fn process_instruction(
    _program_id: &Pubkey,
    keyed_accounts: &[KeyedAccount],    <<<--- this
    data: &[u8],
) -> Result<(), InstructionError> {
```